src/Clb.hs

```diff
@@ -118,51 +95,65 @@ import PlutusLedgerApi.V1 qualified as PV1
 import PlutusLedgerApi.V2 qualified as PV2
 import Prettyprinter (Doc, Pretty, colon, fillSep, hang, indent, pretty, vcat, vsep, (<+>))
 import Test.Cardano.Ledger.Core.KeyPair qualified as TL
-import Text.Show.Pretty (ppShow)
+import qualified Ouroboros.Network.Protocol.LocalStateQuery.Type as Ouroboros
+
+import Cardano.Testnet
+
+import qualified Hedgehog as H
+import qualified Control.Concurrent.STM as STM
+
+import Testnet.Property.Utils (integrationWorkspace)
+import Control.Concurrent (threadDelay, ThreadId)
+import Control.Monad.Trans.Resource
+import qualified Hedgehog.Extras.Stock as H'
+import Testnet.Runtime
+    ( PaymentKeyInfo(paymentKeyInfoPair),
+      PaymentKeyPair(paymentVKey, paymentSKey),
+      PoolNode(poolRuntime),
+      TestnetRuntime(..), getStartTime, shelleyGenesis )
+import Control.Exception (Exception, throwIO)
+import qualified Data.Set as Set
+import Ouroboros.Network.Protocol.LocalTxSubmission.Type (SubmitResult(..))
+import qualified Cardano.Api.InMode as C
 
 --------------------------------------------------------------------------------
 -- Base emulator types
 --------------------------------------------------------------------------------
 
+type KeyPair = TL.KeyPair L.Payment (Core.EraCrypto EmulatorEra)
+
+data ClbRuntime = ClbRuntime
+  { nodeConnectInfo :: Api.LocalNodeConnectInfo
+  , wallets         :: [(FilePath, FilePath)]
+  , slotConfig      :: SlotConfig
+  , threadId        :: ThreadId
+  }
+
 -- FIXME: legacy types converted into type synonyms
 type CardanoTx = Core.Tx EmulatorEra
 type ValidationError = L.ApplyTxError EmulatorEra
 
 data ValidationResult
   = -- | A transaction failed to be validated by Ledger
     Fail !CardanoTx !ValidationError
-  | Success !EmulatedLedgerState !OnChainTx
+  | Success !OnChainTx
   deriving stock (Show)
 
 --------------------------------------------------------------------------------
 -- CLB base monad (instead of PSM's Run)
 --------------------------------------------------------------------------------
 
-newtype ClbT m a = ClbT {unwrapClbT :: StateT ClbState m a}
+newtype Clb a = Clb {unwrapClb :: ReaderT ClbRuntime (WriterT (Log LogEntry) IO) a }
   deriving newtype
     ( Functor
     , Applicative
     , Monad
     , MonadIO
     , MonadFail
-    , MonadState ClbState
+    , MonadReader ClbRuntime
+    , MonadWriter (Log LogEntry)
     )
 
--- | State monad wrapper to run emulator.
-type Clb a = ClbT Identity a
-
-{- | Emulator state: ledger state + some additional things
-FIXME: remove non-state parts like MockConfig and Log (?)
--}
-data ClbState = ClbState
-  { emulatedLedgerState :: !EmulatedLedgerState
-  , mockConfig :: !MockConfig
-  , -- FIXME: rename, this is non-inline dataums cache
-    mockDatums :: !(M.Map P.DatumHash P.Datum)
-  , mockInfo :: !(Log LogEntry)
-  , mockFails :: !(Log FailReason)
```

Comment:
I would suggest we should remove the word `mock` everywhere. There no mock longer exists. RE this particular case, if my memory serves me well, the story was that all error log entries were stored separately in this `Log` structure, and I think I changes that at some point in favor of using `mockInfo` which should have name `emulator(clc)Log`.